### PR TITLE
Refactor the `Hardware` shard

### PR DESCRIPTION
**Pull request by j8r**
* creation of the `Hardware::CPU` module
* creation of the `Hardware::Memory` struct
* improve the API docs in README
* add tests

### CODE REVIEW

File: src/hardware/memory.cr [hunk 1]
```diff
@@ -0,0 +1,34 @@
+struct Hardware::Memory
+  # VmallocTotal can be very huge - needs Int64
+  getter meminfo = Hash(String, Int64).new
+
+  def initialize
+    File.read("/proc/meminfo").each_line do |line|
+      properties = line.split ' '
+      @meminfo[properties.first.rchop] = (properties.last == "kB" ? properties[-2] : properties.last).to_i64
```
Comment (bew) [inline src/hardware/memory.cr]: `properties[-2] * 1024` to keep the result in bytes?

Comment (j8r) [reply src/hardware/memory.cr]: Does it allow to have better performance — is it the only way to do it?

Comment (j8r) [reply src/hardware/memory.cr]: Does it allow to have better performance — is it the only way to do it?

Comment (j8r) [reply src/hardware/memory.cr]: Sorry for the duplicate bug :-/

**Comment by girng** [comment]
working here on WSL beautifully. nice!

**Comment by bararchy** [comment]
Overall this looks good, @j8r did you test it on Ubuntu? 
@girng Thanks for the test on WSL :)

**Comment by bararchy** [comment]
specs passing, Merged!

**Comment by j8r** [comment]
@bararchy Thanks! I've tested on Manjaro Linux. I've tried to test on some Dockers flavours, but the `/proc/meminfo` and `/proc/stat` are quite the same of my host.

**Pull request merged**